pkg/services/auth/authimpl/auth_token.go

```diff
@@ -337,51 +391,76 @@ func (s *UserAuthTokenService) RevokeToken(ctx context.Context, token *auth.User
 }
 
 func (s *UserAuthTokenService) RevokeAllUserTokens(ctx context.Context, userId int64) error {
-	return s.sqlStore.WithDbSession(ctx, func(dbSession *db.Session) error {
-		sql := `DELETE from user_auth_token WHERE user_id = ?`
-		res, err := dbSession.Exec(sql, userId)
+	return s.sqlStore.InTransaction(ctx, func(ctx context.Context) error {
+		ctxLogger := s.log.FromContext(ctx)
+		err := s.sqlStore.WithDbSession(ctx, func(dbSession *db.Session) error {
+			sql := `DELETE from user_auth_token WHERE user_id = ?`
+			res, err := dbSession.Exec(sql, userId)
+			if err != nil {
+				return err
+			}
+
+			affected, err := res.RowsAffected()
+			if err != nil {
+				return err
+			}
+
+			ctxLogger.Debug("All user tokens for user revoked", "userID", userId, "count", affected)
+
+			return nil
+		})
 		if err != nil {
 			return err
 		}
 
-		affected, err := res.RowsAffected()
+		err = s.externalSessionStore.DeleteExternalSessionsByUserID(ctx, userId)
 		if err != nil {
-			return err
+			// Intentionally not returning error here, as the token has been revoked -> the backround job will clean up orphaned external sessions
```

Comment:
I think we should return `err` here because the delete operation for the external sessions runs inside a transaction. If it fails then the user_auth_token records will also not be deleted, they are part of the DB transaction.

After looking closer, I noticed that the `err` is returned later but not inside `if err != nil`. I think it would be more clear if we return err from here.

Reply:
Thanks, you spotted a mistake :). At the end, this inner function should return `nil` not the `err`. I fixed this and also added tests to verify that the tokens are revoked regardless of the result of the `DeleteExternalSessionsByUserID` invocation.